python_ta/transforms/ExprWrapper.py

```diff
@@ -0,0 +1,139 @@
+            elif isinstance(node, nodes.Compare):
+                node = self.parse_compare(node)
+            elif isinstance(node, nodes.BinOp):
+                node = self.parse_bin_op(node)
+            elif isinstance(node, nodes.Const):
+                node = node.value
+            elif isinstance(node, nodes.Name):
+                node = self.apply_name(node.name, self.types[node.name])
+            else:
+                node = None
+
+            return node
+        except Exception:
+            return None
+
+    def apply_name(self, name: str, typ: str) -> Optional[z3.ExprRef]:
+        """Set up the appropriate variable representation in Z3 based on name and type.
+        If an error is encountered or a case is unconsidered, return None."""
+        # TODO: determine full list of supported types
+        if typ == "int":
+            return z3.Int(name)
+        if typ == "float":
+            return z3.Real(name)
+        if typ == "bool":
+            return z3.Bool(name)
+
+        return None
+
+    def parse_compare(self, node: astroid.Compare) -> Optional[z3.ExprRef]:
+        left, ops = node.left, node.ops
+        # ERROR: left, right can be None at any point in the execution
+        left = self.reduce(left)
+        for item in ops:
+            op, right = item
+            right = self.reduce(right)
+            left = self.apply_bin_op(left, op, right)
+        return left
+
+    def apply_unary_op(self, left, op) -> Optional[z3.ExprRef]:
+        # ERROR: left can be None
+        # ERROR: see apply_bool_op ERROR comments
+        if left is None:
+            return None
+        try:
+            if op == "not":
+                return z3.Not(left)
+            else:
+                return None
+        except z3.z3types.Z3Exception:
+            return None
+
+    def apply_bin_op(self, left, op, right) -> Optional[z3.ExprRef]:
+        """Given left, right, op, apply the binary operation."""
+        # ERROR: left or right can be None
+        if left is None or right is None:
+            return None
+        # TODO: find out which binary operations are supported
+        # ERROR: unsupported operation ie. bool + bool (raises TypeError)
+        try:
+            if op == "+":
+                return left + right
+            elif op == "-":
+                return left - right
+            elif op == "*":
+                return left * right
+            elif op == "/":
+                return left / right
+            elif op == "**":
+                return left**right
+            elif op == "==":
+                return left == right
+            elif op == "<=":
+                return left <= right
+            elif op == ">=":
+                return left >= right
+            elif op == "<":
+                return left < right
+            elif op == ">":
+                return left > right
+            else:
+                return None
+        except TypeError:
```

Comment:
Okay, so here I think it would be good to raise a custom exception from inside the `except`. E.g. `raise Z3CustomException(f'variables of incompatible types {left} and {right} for operation {op}')`